### ISSUE TITLE
[Accessibility]: Notepad screen reader issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#36510]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34198

### Problem description

- On VoiceOver on macOS, When in the listbox in the notepad, can't use Ctrl+Opt+Left/right to move between components inside the listbox item. Only up/down arrow keys work and they move focus between the lines, but it's impossible to focus the buttons inside the listbox items.
- Confirmation message not read/focused when clicking on "Remove all notes" button.

### Expected behavior

VoiceOver navigation works as expected and confirmation message is read to users and confirm button is focused after clicking the remove button.

### Additional details

@fkling Let me know if you can help fix this. If you still don't have a Mac device where you can run voiceover, we can reassign.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by fkling** [comment]
I got my MacBook yesterday, so I can look into this 😃

**Comment by limitedmage** [comment]
Perfect timing :D

**Comment by fkling** [comment]
For my understanding: Where does the `Ctrl+Opt+Left/Right` requirement come from? What is it generally supposed to do?

**Comment by limitedmage** [comment]
@fkling These are the keyboard shortcuts that VoiceOver uses to navigate (you don't need to implement this).

**Comment by limitedmage** [comment]
I think we need to remove `tabindex=0` from the notepad items, that seems to be conflicting with a number of things, including screen reader focus.
